parsec/core/mountpoint/winfsp_runner.py

```diff
@@ -110,23 +88,26 @@ async def winfsp_mountpoint_runner(
     trio_token = trio.hazmat.current_trio_token()
     fs_access = ThreadFSAccess(trio_token, workspace_fs)
 
-    mountpoint_path = await _bootstrap_mountpoint(base_mountpoint_path, workspace_name)
+    user_manifest = user_fs.get_user_manifest()
+    workspace_ids = [entry.id for entry in user_manifest.workspaces]
+    workspace_index = workspace_ids.index(workspace_fs.workspace_id)
+    mountpoint_path = await _get_available_drive(workspace_index, len(workspace_ids))
 
     if config.get("debug", False):
         enable_debug_log()
 
     # Volume label is limited to 32 WCHAR characters, so force the label to
     # ascii to easily enforce the size.
     volume_label = (
-        unicodedata.normalize("NFKD", f"{device.user_id}-{workspace_name}")
+        unicodedata.normalize("NFKD", f"{workspace_name.capitalize()}")
         .encode("ascii", "ignore")[:32]
         .decode("ascii")
     )
     volume_serial_number = _generate_volume_serial_number(device, workspace_fs.workspace_id)
     operations = WinFSPOperations(event_bus, volume_label, fs_access)
     # See https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getvolumeinformationa  # noqa
     fs = FileSystem(
-        str(mountpoint_path.absolute()),
+        mountpoint_path.drive,
```

Comment:
What happens if the drive has been taken in the meantime ?

Reply:
At the moment, the mount task will simply fail with a `MountpointDriverCrash` exception.

Should we investigate a better solution?

Reply:
no I think it's fine (as long as the GUI can handle this properly of course ! ^^)